server/cmd/mattermost/commands/export.go

```diff
@@ -134,23 +134,22 @@ func scheduleExportCmdF(command *cobra.Command, args []string) error {
 		ctx := context.Background()
 		if timeoutSeconds > 0 {
 			var cancel context.CancelFunc
-			ctx, cancel = context.WithTimeout(ctx, time.Second*time.Duration(timeoutSeconds))
+			_, cancel = context.WithTimeout(ctx, time.Second*time.Duration(timeoutSeconds))
 			defer cancel()
 		}
 
-		var rctx request.CTX = request.EmptyContext(a.Log())
-		rctx = rctx.WithContext(ctx)
```

Comment:
Why was `WithContext` removed?

Reply:
Initially the `ctx` context was declared and assigned with `context.Background()` which is just an empty context. This means that creating a context with `request.EmptyContext(a.Log())` and `request.EmptyContext(a.Log()).WithContext(ctx)` is the same thing to me

Reply:
In L137 a new `ctx` gets created using a timeout. Why is that `ctx` not longer needed?

Reply:
Alright I saw that that is done inside the if statement. To me the new ctx which created with timeout is local to the if block.

Reply:
Or should I bring back the line
```
rctx = rctx.WithContext(ctx)
```
as before?

Reply:
Hello @hanzei, I've added the line of code `rctx = rctx.WithContext(ctx)`

Note that I had to change declaration at the top to `var rctx request.CTX = request.EmptyContext(a.Log())`. This is because using `rctx := request.EmptyContext(a.Log())` automatically gives `rctx` a type of `request.Context`. When I added the line 
```
rctx = rctx.WithContext(ctx)
```
I got an error because `rctx.WithContext(ctx)` returns a value of type `*request.CTX` but `rctx` given type `*request.Context`